frontend/src/screens/ProjectEditor.tsx

```diff
@@ -1,162 +1,1261 @@
+                    onClick={() => addFile("css")}
+                  >
+                    CSS
+                  </Menu.Item>
+                  <Menu.Item
+                    leftSection={<PiFileJs size={14} />}
+                    onClick={() => addFile("js")}
+                  >
+                    JavaScript
+                  </Menu.Item>
+                </Menu.Dropdown>
+              </Menu>
+              <ActionIcon
+                variant="subtle"
+                onClick={() => closePane("explorer")}
+                size="sm"
+              >
+                <PiXBold />
+              </ActionIcon>
+            </Group>
+            <Box style={{ flex: 1, overflowY: "auto" }}>
+              {projectFiles.map((file) => (
+                <Box
+                  key={file.fileName}
+                  px="sm"
+                  py={6}
+                  style={{
+                    cursor: "pointer",
+                    background:
+                      selectedFile === file.fileName ? "#f3f3f3" : undefined,
+                    fontWeight: selectedFile === file.fileName ? 600 : 400,
+                    display: "flex",
+                    alignItems: "center"
+                  }}
+                  onClick={() => handleFileSelect(file.fileName)}
+                  onDoubleClick={() => startRename(file.fileName)}
+                >
+                  {unsavedFiles[file.fileName] && (
+                    <PiDotOutlineFill
+                      color="#FFA94D"
+                      style={{ marginRight: 6, fontSize: 16 }}
+                    />
+                  )}
+                  {renamingFile === file.fileName ? (
+                    <TextInput
+                      value={renameValue}
+                      onChange={(e) =>
+                        dispatch(setRenameValue(e.currentTarget.value))
+                      }
+                      size="xs"
+                      autoFocus
+                      onBlur={confirmRename}
+                      onKeyDown={(e) => {
+                        if (e.key === "Enter") {
+                          confirmRename();
+                        } else if (e.key === "Escape") {
+                          cancelRename();
+                        }
+                      }}
+                      styles={{
+                        input: {
+                          padding: "2px 6px",
+                          fontSize: "14px",
+                          height: "24px",
+                          minWidth: "80px"
+                        }
+                      }}
+                      onClick={(e) => e.stopPropagation()}
+                    />
+                  ) : (
+                    file.fileName
+                  )}
+                </Box>
+              ))}
+            </Box>
+          </Paper>
+        );
+      case "editor":
+        const file = projectFiles.find((f) => f.fileName === activeTab) || {

```

Comment:
similar to above, and also applied to all below - these can be their own components